sound/soc/codecs/rt1320-sdw.c

```diff
@@ -0,0 +1,1678 @@
+	{ 0x10007655, 0xd0 },
+	{ 0x10007656, 0x0f },
+	{ 0x10007657, 0xa3 },
+	{ 0x10007658, 0x83 },
+	{ 0x10007659, 0xc7 },
+	{ 0x1000765a, 0x11 },
+	{ 0x1000765b, 0x42 },
+	{ 0x1000765c, 0x63 },
+	{ 0x1000765d, 0x86 },
+	{ 0x1000765e, 0x07 },
+	{ 0x1000765f, 0x00 },
+	{ 0x10007660, 0x03 },
+	{ 0x10007661, 0xc7 },
+	{ 0x10007662, 0x01 },
+	{ 0x10007663, 0x42 },
+	{ 0x10007664, 0x63 },
+	{ 0x10007665, 0x10 },
+	{ 0x10007666, 0x07 },
+	{ 0x10007667, 0x02 },
+	{ 0x10007668, 0x83 },
+	{ 0x10007669, 0xc6 },
+	{ 0x1000766a, 0x21 },
+	{ 0x1000766b, 0x41 },
+	{ 0x1000766c, 0x13 },
+	{ 0x1000766d, 0x07 },
+	{ 0x1000766e, 0xf0 },
+	{ 0x1000766f, 0x01 },
+	{ 0x10007670, 0x13 },
+	{ 0x10007671, 0x05 },
+	{ 0x10007672, 0xf0 },
+	{ 0x10007673, 0x01 },
+	{ 0x10007674, 0x63 },
+	{ 0x10007675, 0x98 },
+	{ 0x10007676, 0xe6 },
+	{ 0x10007677, 0x02 },
+	{ 0x10007678, 0x63 },
+	{ 0x10007679, 0x8a },
+	{ 0x1000767a, 0x07 },
+	{ 0x1000767b, 0x02 },
+	{ 0x1000767c, 0x83 },
+	{ 0x1000767d, 0xc7 },
+	{ 0x1000767e, 0x01 },
+	{ 0x1000767f, 0x42 },
+	{ 0x10007680, 0x63 },
+	{ 0x10007681, 0x86 },
+	{ 0x10007682, 0x07 },
+	{ 0x10007683, 0x02 },
+	{ 0x10007684, 0x83 },
+	{ 0x10007685, 0xc7 },
+	{ 0x10007686, 0x31 },
+	{ 0x10007687, 0x42 },
+	{ 0x10007688, 0x63 },
+	{ 0x10007689, 0x86 },
+	{ 0x1000768a, 0x07 },
+	{ 0x1000768b, 0x00 },
+	{ 0x1000768c, 0x83 },
+	{ 0x1000768d, 0xc7 },
+	{ 0x1000768e, 0x21 },
+	{ 0x1000768f, 0x42 },
+	{ 0x10007690, 0x63 },
+	{ 0x10007691, 0x9e },
+	{ 0x10007692, 0x07 },
+	{ 0x10007693, 0x00 },
+	{ 0x10007694, 0x03 },
+	{ 0x10007695, 0xc7 },
+	{ 0x10007696, 0x21 },
+	{ 0x10007697, 0x41 },
+	{ 0x10007698, 0x93 },
+	{ 0x10007699, 0x07 },
+	{ 0x1000769a, 0xb0 },
+	{ 0x1000769b, 0x01 },
+	{ 0x1000769c, 0x63 },
+	{ 0x1000769d, 0x08 },
+	{ 0x1000769e, 0xf7 },
+	{ 0x1000769f, 0x00 },
+	{ 0x100076a0, 0x13 },
+	{ 0x100076a1, 0x05 },
+	{ 0x100076a2, 0xb0 },
+	{ 0x100076a3, 0x01 },
+	{ 0x100076a4, 0xef },
+	{ 0x100076a5, 0xd0 },
+	{ 0x100076a6, 0x8f },
+	{ 0x100076a7, 0xeb },
+	{ 0x100076a8, 0xef },
+	{ 0x100076a9, 0xd0 },
+	{ 0x100076aa, 0x0f },
+	{ 0x100076ab, 0xc1 },
+	{ 0x100076ac, 0x83 },
+	{ 0x100076ad, 0x20 },
+	{ 0x100076ae, 0xc1 },
+	{ 0x100076af, 0x00 },
+	{ 0x100076b0, 0x13 },
+	{ 0x100076b1, 0x07 },
+	{ 0x100076b2, 0x10 },
+	{ 0x100076b3, 0x00 },
+	{ 0x100076b4, 0xa3 },
+	{ 0x100076b5, 0x89 },
+	{ 0x100076b6, 0xe1 },
+	{ 0x100076b7, 0x40 },
+	{ 0x100076b8, 0x13 },
+	{ 0x100076b9, 0x01 },
+	{ 0x100076ba, 0x01 },
+	{ 0x100076bb, 0x01 },
+	{ 0x100076bc, 0x67 },
+	{ 0x100076bd, 0x80 },
+	{ 0x100076be, 0x00 },
+	{ 0x100076bf, 0x00 },
+	{ 0x3fc2bfc7, 0x00 },
+	{ 0x3fc2bfc6, 0x00 },
+	{ 0x3fc2bfc5, 0x00 },
+	{ 0x3fc2bfc4, 0x01 },
+	{ 0x0000d486, 0x43 },
+	{ 0x1000db00, 0x02 },
+	{ 0x1000db01, 0x00 },
+	{ 0x1000db02, 0x11 },
+	{ 0x1000db03, 0x00 },
+	{ 0x1000db04, 0x00 },
+	{ 0x1000db05, 0x82 },
+	{ 0x1000db06, 0x04 },
+	{ 0x1000db07, 0xf1 },
+	{ 0x1000db08, 0x00 },
+	{ 0x1000db09, 0x00 },
+	{ 0x1000db0a, 0x40 },
+	{ 0x0000d540, 0x01 },
+};
```

Comment:
Ah this is fantastic timing, I was just working on "blind writes".

@shumingfan can you take a look at this PR https://github.com/thesofproject/linux/pull/4930/commits/20eb0aa7af2a59e59ec8b609fcffa06c8a464e59 and comment on whether you could use the mipi-sdca-initialization-table if it's available?

Reply:
@plbossart will try

Reply:
@shumingfan do you know if this table is always the same to make the codec work or if it contains variants for different platform configurations. That would change the way we look at this initialization table.

Reply:
@plbossart I think it might be changed for different boards. 
If the codec changes the HW version, the initialization table might also change.

Reply:
ok so it would help to have regmap defaults in the driver, and if the table is present then the driver can update the defaults based on the table values to match what the board/version requires.